src/lib/containers/widgets/facet-nav/FacetNavPanel.tsx

```diff
@@ -108,58 +102,76 @@ export function extractPlotDataArray(
     facetToPlot.facetValues.length,
   )
 
-  const getLabels = (
+  const getLabels = async (
     facetValues: FacetColumnResultValueCount[],
-    truncateFlag: boolean,
     columnType?: ColumnType,
+    sessionToken?: string,
   ) => {
+    const map = new Map<string, string>()
+    map.set(
+      SynapseConstants.VALUE_NOT_SET,
+      SynapseConstants.FRIENDLY_VALUE_NOT_SET,
+    )
+    // Filter out empties
+    const filteredValues = facetValues
+      .map(value => value.value)
+      .filter(val => val !== SynapseConstants.VALUE_NOT_SET)
+    if (columnType === ColumnType.ENTITYID) {
+      // TODO: Pagination
+      const response = await SynapseClient.getEntityHeadersByIds(
+        filteredValues,
+        sessionToken,
+      )
+      for (const header of response.results) {
+        map.set(header.id, header.name)
+      }
+    } else if (columnType === ColumnType.USERID) {
+      const response = await SynapseClient.getGroupHeadersBatch(
+        filteredValues,
+        sessionToken,
+      )
```

Comment:
Both of these requests return the deprecated `PaginatedList` response that possibly give an inaccurate `totalNumberOfResults` based on offset, but the offset is unknown because failures are omitted. There's no offset parameter on the service, so we can't just get the next page.

To get all results reliably, we may have to do our own reconciliation where we double-request entities to see if they were omitted because of a failure or because of pagination.

I don't think it's a huge deal to not paginate here since we make requests in descending order by count (so the largest pie slices will be requested first).

[Filed a backend ticket here](https://sagebionetworks.jira.com/browse/PLFM-6714).